### ISSUE TITLE
Toggle Off Shift while drawing.

### ISSUE THREAD

**Issue by AguRex7** [tahoma2d/tahoma2d#672]
**Is your feature request related to a problem? Please describe**
A clear and concise description of what the problem is. Ex. I'm always frustrated when I draw and accidentally tap shift and draws a straight line. I would like the assist feature on the brush be toggled odd and on.

**Describe the solution you'd like**

**Comment by RattusaurusRex** [comment]
I would also like to be able to toggle the Assist feature on and off.  I would sometimes get a straight line after using Undo, even when Ctrl+Z is mapped to an express key (which means I'm not accidentally holding my finger down on the Ctrl key).

**Comment by AguRex7** [comment]
Yeah... Exactly. It's Ctrl not Shift. And it also happens when I us Ctrl Z. I thought I was accidentally pressing the Ctrl Key

**Comment by artisteacher** [comment]
Holding Ctrl when drawing locks the line to 15deg? angles.

On macOS, as long as I finish drawing my line before I use the undo command, I can't incorrectly trigger a straight line from using undo. If I try to use undo while actively drawing a line, then the last stroke (or etc.) will undo and the in progress stroke goes straight. Personally, I never use undo in the midst of drawing a line. 

Perhaps some of the drawing assist features should require a long hold to be triggered...

**Comment by DarrenTAnims** [comment]
I've had this "ghost" ctrl issue for a while, but haven't been able to pin down why exactly it happens. But it happens so often that I've actually remapped my undo to just be Z. Sometime after drawing, then performing an undo, when I start to draw, it draws a straight line as though undo is pressed, which it isn't. To get out of it I have to either hold Ctrl down for a few seconds and release or press it 5 or 6 times (someone suggested I try it and it worked), but it does feel like I'm casting a spell to remove the ghost ctrl hex. WOOOOoooooo...

**Comment by melieconiek** [comment]
This is a legitimate issue but adding any kind noticabable delay for keybindings is a non-starter. I use the shift modifier for the brush tool *a lot*. I need it to work immediately so I can move on.

**Comment by manongjohn** [comment]
I suspect there might be an issue with some internal stuff not being tracked correctly when pressing/releasing the CTRL key.  Part of that also depends on when you are pressing/releasing CTRL. It's hard to tell since I have difficulty trying to recreate this issue. 

If someone can somehow capture the issue on video, preferably if you have an app that shows keys being pressed, I might be able to see better what's going on and maybe recreate it myself too.

**Comment by DarrenTAnims** [comment]
It's recorded on a couple of my videos (I changed my undo command after those). One in particular I remember struggling to turn off the ctrl-lock while I drew. But there's no on screen key app showing what I pressed, but I don't think that would help anyway.
